packages/react-heartwood-components/src/components/Core/components/FeedBuilder/FeedBuilder.js

```diff
@@ -0,0 +1,251 @@
+// @flow
+import React, { Component, Fragment } from 'react'
+import moment from 'moment-timezone'
+import {
+	List,
+	AutoSizer,
+	CellMeasurer,
+	CellMeasurerCache,
+	InfiniteLoader
+} from 'react-virtualized'
+import Message, { MessageBuilder } from '../../../Message'
+import Text from '../../../Text/Text'
+
+import type from '../../../Message'
+
+type MessageProps = {
+	/** Unique id for the message */
+	id: string,
+	...MessageBuilderProps
+}
+
+type Props = {
+	/** Messages for the feed */
+	messages?: Array<MessageProps>,
+
+	/** The total number of messages in the feed. Used for infinite scrolling. */
+	messageCount?: number,
```

Comment:
Is this necessary? For infinite loading, wouldn't you just stop loading if you went back far enough and time and got zero results? I actually thought that was the default behavior of the virtualized loader component

Reply:
@gcpantazis I added this to make scrolling up smoother when new items are added. It sets the `scrollToIndex` to keep the position consistent when new items load in. This also prevents the `loadMoreRows` callback from firing over and over again by scrolling down.

Reply:
I feel like this wasn't necessary the last time I made one of these. And plus I don't know that we can rely on being able to figure out how many thousands of items might be included in some feeds... and chats get to the thousands of entries very very quickly.

Reply:
@jtini I took a look at my last project that used this, and it seemed like I handled this by `return`ing out of `loadMoreRows` if I was already loading something (which in that case I tracked by setting `isLoading` to true on state). This worked really nicely and let me load infinitely backwards without knowing for certain how far back the history went